src/script/finale.ts

```diff
@@ -23,19 +46,9 @@ function generateDocument(query: string) {
             if (xhr.readyState === 4) {
                 if (xhr.status === 200) {
                     if (xhr.responseText) {
-                        const parsedContent = Model.parseJSON(xhr, originPath);
-                        const viewObj = new ViewFactory();
-                        const htmlDOM = viewObj.generate(parsedContent);
-                        if (htmlDOM) {
-                            const url = document.location.search + document.location.hash;
-                            HistoryManager.UpdateHistory(url, parsedContent);
-                            const finale = document.getElementById("divLog");
-                            while (finale.hasChildNodes()) {
-                                finale.removeChild(finale.lastChild);
-                            }
-                            finale.appendChild(htmlDOM);
-                            top.postMessage({ highlight: true, id: query.slice(3).trim() }, "*");
-                        }
+                        parsedContent = Model.parseJSON(xhr, originPath);
+                        sessionStorage.setItem(document.location.search.slice(4), JSON.stringify(parsedContent));
```

Comment:
Saving the DOM at this state always saves state at the point when it is generated, not at the state the user would have left it at.

So basically, the purpose of saving state is to remember the VI (or any other file) exactly how we left it at the most recently used time. For example, if I switch to FrontPanel, then click on some other file on the file explorer, and come back to the first file, I need it to open at FrontPanel. Saving the state at this point in time stores the VI at its generated state, doesn't it? How exactly does sessionStorage.setItem() work? Does it stringify the parsedContent at this point of code execution or at a later point, when the DOM is getting replaced by a newly generated DOM?

Reply:
Fixed, closing this comment